src/gensxg.cc

```diff
@@ -0,0 +1,307 @@
+#include <time.h>
+#include <unistd.h>
+
+#include <openssl/pem.h>
+
+#include <memory>
+#include <string>
+#include <iostream>
+
+#include "libsxg.h"
+
+struct Options {
```

Comment:
This looks more like a class, if you keep using C++. ([style guide](https://google.github.io/styleguide/cppguide.html#Structs_vs._Classes))